### PR TITLE
Add common methods to weatherobject

### PR DESCRIPTION
**Pull request by rejas**
Looking at the different weather providers I noticed some methods were repeated.

So I refactored some of them into the weatherobject to be used from the providers.

Added a unit test too, so hopefully it is correct. But better to have more than one pair of eyes look at the changes since the providers are hard to test :-)

Question though: Some wind / temp conversions seem to be doubled in there too, should I take a look at them too?

### CODE REVIEW

**Comment by khassel** [comment]
test errors may caused by `node-fetch`: https://github.com/node-fetch/node-fetch/blob/main/docs/v3-UPGRADE-GUIDE.md
Will take a look at this.

File: modules/default/weather/weatherobject.js [hunk 1]
```diff
@@ -10,6 +12,13 @@
  * As soon as we start implementing the forecast, mode properties will be added.
  */
 class WeatherObject {
+	/**
+	 *
+	 * @param units
```
Comment (khassel) [inline modules/default/weather/weatherobject.js:17]: what about all the following warnings `Missing JSDoc @param ...`?

Comment (rejas) [reply modules/default/weather/weatherobject.js:17]: yeah, will fix those with the next commits over the weekend. stay tuned :-)

**Comment by MichMich** [comment]
Thanks! I love refactors like these!

**Comment by khassel** [comment]
@rejas after this was merged I'm getting errors in one unit test.

I know these tests passed here but I'm running the tests additionally after building the docker images (here the develop image).

Tested now locally and with my `TZ=Europe/Berlin` everything works fine.

But if I set e.g. `TZ=America/New_York` the unit tests fail:

```bash
 FAIL   unit  tests/unit/functions/weather_object_spec.js
  WeatherObject
    ✕ should return true for daytime at noon (2 ms)
    ✕ should return false for daytime at midnight (1 ms)

  ● WeatherObject › should return true for daytime at noon

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      14 |              weatherobject.date = moment(12, "HH");
      15 |              weatherobject.updateSunTime(-6.774877582342688, 37.63345667023327);
    > 16 |              expect(weatherobject.isDayTime()).toBe(true);
         |                                                ^
      17 |      });
      18 |
      19 |      it("should return false for daytime at midnight", function () {

      at Object.<anonymous> (tests/unit/functions/weather_object_spec.js:16:37)

  ● WeatherObject › should return false for daytime at midnight

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      20 |              weatherobject.date = moment(0, "HH");
      21 |              weatherobject.updateSunTime(-6.774877582342688, 37.63345667023327);
    > 22 |              expect(weatherobject.isDayTime()).toBe(false);
         |                                                ^
      23 |      });
      24 | });
      25 |

      at Object.<anonymous> (tests/unit/functions/weather_object_spec.js:22:37)
```

So it seems these tests are now depending on which timezone they are running.

**Comment by rejas** [comment]
> @rejas after this was merged I'm getting errors in one unit test.

Yes, I can confirm that when changing my tz too. ALready have a fix for that, question is @MichMich, should I make a seperate PR for that fix or can I put it in https://github.com/MichMich/MagicMirror/pull/2650 ?

**Comment by MichMich** [comment]
Just put it in the same. That's ok.

**Comment by rejas** [comment]
Done :-)

**Pull request merged**